### ISSUE TITLE
[BUG] Incompatibility with hypothesis 6.41.0

### ISSUE THREAD

**Issue by mdantonio** [schemathesis/schemathesis#1425]
**Checklist**

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)

**Describe the bug**
The latest (just released) version of hypothesis (6.41.0) is not compatible
```
  File "/usr/local/lib/python3.9/dist-packages/schemathesis/utils.py", line 34, in <module>
    from hypothesis.utils.conventions import InferType
ImportError: cannot import name 'InferType' from 'hypothesis.utils.conventions' (/usr/local/lib/python3.9/dist-packages/hypothesis/utils/conventions.py)
```

**To Reproduce**
1. pip3 install schemathesis
2. import schemathesis
3. See error

**Expected behavior**
It is expected a compatibility with hypothesis 6.41.0 or the hypothesis version pinned to <6.41.0

**Environment (please complete the following information):**
 - OS: Linux
 - Python version: 3.9.5
 - Schemathesis version: 3.13.6
- hypothesis version: 6.41.0

**Comment by Stranger6667** [comment]
Thanks for reporting! Will fix soon :)

**Comment by Stranger6667** [comment]
The fix is released in `3.13.7`

**Comment by mdantonio** [comment]
Thank you!
